addons/sale/views/product_attribute_views.xml

```diff
@@ -32,11 +32,13 @@
         <field name="priority" eval="8"/>
         <field name="arch" type="xml">
             <xpath expr="//field[@name='create_variant']" position="before">
-                <field name="type" widget="radio"/>
+                <field name="display_type" widget="radio"/>
             </xpath>
             <xpath expr="//field[@name='value_ids']/tree/field[@name='name']" position="after">
-                <field name="is_custom" groups="product.group_product_variant" />
-                <field name="html_color"/>
+                <field name="display_type" invisible="1"/>
+                <field name="is_custom" groups="product.group_product_variant"/>
+                <field name="html_color" widget="color"
+                       attrs="{'column_invisible': [('parent.display_type','in',('select', 'radio'))]}"/>
```

Comment:
Nitpicking: xml domains are evaluated in Python, in which it is recommended to add spaces after commas in iterable data types.